ligo/src/types.mligo

```diff
@@ -0,0 +1,230 @@
+// Corresponds to Types.hs module
+
+// -- FA2 types -- //
+
+#if !TYPES_H
+#define TYPES_H
+
+type token_id = nat
+type token_metadata =
+  [@layout:comb]
+  { token_id : token_id
+  ; symbol : string
+  ; name : string
+  ; decimals : nat
+  ; extras : (string, string) map
+  }
+
+type operator =
+  { owner : address
+  ; operator : address
+  }
+type operators = (operator, unit) big_map
+
+type ledger_key = address * token_id
+type ledger_value = nat
+type ledger = (ledger_key, ledger_value) big_map
+
+// -- DAO base types -- //
+
+type migration_status =
+  [@layout:comb]
+  | Not_in_migration
+  | MigratingTo of address
+  | MigratedTo of address
+
+type proposal_key = bytes
+type proposal_metadata = (string, bytes) map
+type proposal =
+  [@layout:comb]
+  { upvotes : nat
+  ; downvotes : nat
+  ; start_date : timestamp
+  ; metadata : proposal_metadata
+  ; proposer : address
+  ; proposer_frozen_token : nat
+  ; voters : (address * nat) list
+  }
+
+type vote_type = bool
+
+type permit =
+  { key : key
+  ; signature : signature
+  }
+
+type contract_extra = (string, bytes) map
+
+// -- Storage -- //
+
+type storage =
+  { ledger : ledger
+  ; operators : operators
+  ; token_address : address
+  ; admin : address
+  ; pending_owner : address
+  ; migration_status : migration_status
+  ; voting_period : nat
+  ; quorum_threshold : nat
+  ; extra : contract_extra
+  ; proposals : (proposal_key, proposal) big_map
+  ; proposals_key_list_sort_by_date : proposal_key list
+  ; permits_counter : nat
+  }
+
+// -- Parameter -- //
+
+type transfer_destination =
```

Comment:
Even tho this is indeed part of the parameter, it's also part of FA2's interface for which there is a "group" above, do you think it would make sense to move this (and all the following types up to `fa2_parameter`) under the `FA2 types` banner?

Reply:
Yes, this was a questionable point. Since this type is a dedicated parameter part, should I also move `fa2_parameter` above?

Reply:
Not sure, I'm more for the "yes" only because being part of FA2 is more specific than being part of the parameter, but his is indeed subjective.

Reply:
I thought I had replied to this, but apparently not.

I was for moving that too anyway and it seems you have done so as well, so all good :slightly_smiling_face: